include/zephyr/arch/arm/asm_inline_gcc.h

```diff
@@ -61,7 +61,15 @@ static ALWAYS_INLINE unsigned int arch_irq_lock(void)
 		"mrs %0, BASEPRI;"
 		"msr BASEPRI_MAX, %1;"
 		"isb;"
-		: "=r"(key), "=r"(tmp)
+		: "=r"(key),
+#if defined(CONFIG_ARMV8_M_BASELINE)
```

Comment:
Is that possible we use `: "=r"(key), "=l"(tmp)` instead of adding a guard?

Reply:
It would work but overconstrains the register lists on most targets. Wouldn't that negatively impact the optimisation oportinuties for the compiler?

Reply:
Yes, this is reasonable. I was just raising an open question to see if there is any chance to reduce the guards in aarch32 in the future. And definitely, this is not the work of this PR. :)